### ISSUE TITLE
Twitter: better error handling?

### ISSUE THREAD

**Issue by ajdapretnar** [biolab/orange3-text#805]
**Describe the bug**
It is known Twitter API doesn't have very distinct error handling. But perhaps there's room for improvement.

One error I get is running Author search on non-existing username (i.e. orange).
I get
<img width="420" alt="Screenshot 2022-03-29 at 10 01 13" src="https://user-images.githubusercontent.com/12524972/160563605-81db1e77-149d-46d0-ac5c-410c3134ce0e.png">

Running Author search on existing username with full handle (i.e. @OrangeDataMiner) returns an error:
<img width="614" alt="Screenshot 2022-03-29 at 10 01 33" src="https://user-images.githubusercontent.com/12524972/160563699-11d250ca-f2ea-4490-a3cc-e048a8814e2a.png">

**To Reproduce**
1. Twitter.
2. Run Author search with "orange".
3. Run Author search with "@OrangeDataMiner".

**Expected behavior**
Search is handled gracefully. Full handles are accepted (they were before the new Twitter fix). When username is not valid (or other API error appears), perhaps we can provide more informative errors. If not, feel free to close the issue.

**Orange version:**
3.32.dev

**Text add-on version:**
1.7.0

**Comment by ajdapretnar** [comment]
TL;DR: go through different types of error and see whether it is possible to show a better error message.
Types of errors: Author not found, run Author with @handle, query contain invalid characters, query empty (what happens), rate limit exceeded, invalid bearer token (perhaps this one is already handled)...

**Comment by needNewUsername** [comment]
It seems to me, the first error is already handled.
![image](https://user-images.githubusercontent.com/27441731/162920591-0e87a1a2-1d44-4d1c-a420-d72e63bf414c.png)
And the second error could be fixed by adding something like this
`def remove_at(word_list):
    return [word[1:] if word[0] == "@" else word for word in word_list]`
but at that point I'm wondering how many odd inputs the widget should accept and try to interpret before throwing the same error?

**Comment by ajdapretnar** [comment]
Indeed, the first error is resolved.

The second could easily be resolved with `"@OrangeDataMiner".strip("@")`, no? Just handle author input at source. The old behavior only supported handles with @ prepended, so in order to keep backward compatibility, this would be urgent. Hence this particular input should be handled.
